src/Temporalio.Extensions.OpenTelemetry/README.md

```diff
@@ -53,6 +53,45 @@ var client = await TemporalClient.ConnectAsync(new("localhost:7233")
 // Now client and worker calls are traced...
 ```
 
+If you are using dependency injection with `Temporalio.Extensions.Hosting`, you have to configure the client and
+worker separately. For example,
+
+```csharp
+// In Program.cs
+using Temporalio.Extensions.Hosting;
+using Temporalio.Extensions.OpenTelemetry;
+
+// Setup the tracer provider
+using var tracerProvider = Sdk.CreateTracerProviderBuilder().
+    AddSource(
+        TracingInterceptor.ClientSource.Name,
+        TracingInterceptor.WorkflowsSource.Name,
+        TracingInterceptor.ActivitiesSource.Name).
+    AddConsoleExporter().
+    Build();
+
+// Setup dependency injection
+var builder = WebApplication.CreateBuilder(args);
+builder.Services
+    // Create a client to localhost on "default" namespace with the tracing
+    // interceptor
+    .AddTemporalClient(opts =>
+        {
+            opts.TargetHost = "localhost:7233";
+            opts.Namespace = "default";
+            opts.Interceptors = new[] { new TracingInterceptor() };
+        })
+    // Create a worker to localhost on "default" namespace with the tracing
+    // interceptor
+    .AddHostedTemporalWorker("localhost:7233", "default", "my-task-queue")
```

Comment:
Per the hosting README, you could use the task-queue-only `AddHostedTemporalWorker` overload and it _will_ use the `ITemporalClient` from DI, which would make more sense in this example (why not reuse the client?). From the README:

> One overload of `AddHostedTemporalWorker`, used in the quick start sample above, accepts the client target host, the client namespace, and the worker task queue. This form will connect to a client for the worker. The other overload of `AddHostedTemporalWorker` only accepts the worker task queue. In the latter, an `ITemporalClient` can either be set on the services container and therefore reused across workers, or the resulting builder can have client options set.

Reply:
Oh! TIL. I thought it was weird that I had to configure the client twice, I must have missed it skimming that README (I'm bad at reading prose 😅 ). Like before, I think it isn't this README's job to explain hosting, but it does make sense to give good examples that replicate real hosting scenarios.

AIUI, the task-queue only overload should be a one-liner, so I think I can fit it into the same code block (perhaps in a comment) + an explanation of when users will need to use which.

Reply:
>  it does make sense to give good examples that replicate real hosting scenarios.

But this is a bad example because you're creating a client twice unnecessarily. Either remove `AddTemporalClient` or change `AddHostedTemporalWorker` to use the overload that uses that client.

> AIUI, the task-queue only overload should be a one-liner, so I think I can fit it into the same code block (perhaps in a comment) + an explanation of when users will need to use which.

Sure. You can show both ways. You can show a client + worker that uses client, and a no-client + worker that creates a client internally. Feel free to get rid of the tracer provider part of the sample as that's likely implied from the previous samples.

Reply:
>> it does make sense to give good examples that replicate real hosting scenarios.
> But this is a bad example because you're creating a client twice unnecessarily. Either remove AddTemporalClient or change AddHostedTemporalWorker to use the overload that uses that client.

Ah yes, that was what I meant by that comment (that my example is not great).

> Feel free to get rid of the tracer provider part of the sample as that's likely implied from the previous samples.

Yeah I was on the fence about this. I'll probably leave a comment to the effect of "set up the tracer like in the previous example".